proposals/291-loeschkonzept-v2.md

```diff
@@ -0,0 +1,369 @@
+
+Anbieter MÜSSEN ihre Nutzer vorab über die serverseitige Löschung von Räumen und
+Rauminhalten informieren. **\[\<=\]**
+
+Weiterhin müssen Nutzer für die Organisation ihrer Unterhaltungen in die Lage
+versetzt werden Räume selbstständig zu verlassen und vom Client zu entfernen.
+Hierbei ist zu beachten, dass Räume nach erfolgreichem [`/leave`] weiterhin in
+der [`/sync`]-Response auftauchen und dort erst durch ein explizites [`/forget`]
+verschwinden. Clients steht es frei diese beiden Operationen zu kombinieren oder
+zu trennen. Werden sie getrennt, entsteht dadurch gewissermaßen eine
+Zwischenablage für historische Räume.
+
+**A_3 - Verlassen von Räumen**
+
+Clients MÜSSEN Nutzern erlauben Räume über die Nutzung der APIs [`/leave`] und
+[`/forget`] vom Client zu löschen. **\[\<=\]**
+
+*Hinweis: Clients steht es frei die Operationen [`/leave`] und [`/forget`] zu
+trennen um dem Nutzer zu ermöglichen ein Archiv verlassener Räume zu erstellen.*
+
+Eine regelmäßige Erinnerung an das clientseitige Löschen von Räumen ist nicht
+erforderlich. Zum Einen ist der überwiegende Teil der Kommunikation in TI-M
+Ende-zu-Ende verschlüsselt. Zum Anderen lässt sich bei mehreren Raumteilnehmern
+von verschiedenen Homeservern vom Client ohnehin kein tatsächliches Löschen der
+Daten auf dem Server durchsetzen. Gleichzeitig steht es Herstellern aber frei
+intelligente Systeme zur Erkennung abgeschlossener Unterhaltungen zu entwickeln
+und diese dem Nutzer zur Löschung anzubieten.
+
+### Redactions
+
+**Story 7**
+
+- Als Endnutzer
+- möchte ich von mir gesendete Nachrichten für alle Gesprächsteilnehmer löschen
+- damit ich schwerwiegende Fehler korrigieren kann.
+
+Die Matrix-Spezifikation ermöglicht die (Selbst-)Moderation von Events mittels
+[Redactions]. Redactions sind eine invasive Form des Löschens da sie über die
+Föderation propagieren und letztendlich zu einer irreversiblen Löschung von
+Inhalten auf allen beteiligten Servern und Clients führen.
+
+Dieses Verhalten ist in bestimmten Fällen wünschenswert. Man stelle sich z. B.
+vor es wurden Daten eines Patienten versehentlich an einen anderen Patienten
+gesendet. Hier erscheint es sinnvoller den Versender diesen Fehler direkt per
+Redaction korrigieren zu lassen anstatt einen langsameren Eskalationsprozess mit
+Löschung durch einen Administrator zu verwenden.
+
+Gleichzeitig führen Redactions bei Fehlbenutzung aber zu einem unerwarteten
+Verlusts von eigentlich relevanten Nachrichten für andere Gesprächsteilnehmer.
+Als Kompromiss werden Redactions daher zwar erlaubt. Sie müssen im Client aber
+stets mit einem Warnhinweis versehen werden[^1].
+
+**A_4 - Nachrichtenbasiertes Löschen per Redaction**
+
+Clients MÜSSEN ihren Nutzern erlauben eigene Nachrichten per Redaction zu
+löschen. Dabei MUSS der Nutzer vor jedem Auslösen einer Redaction per
+Warnhinweis darauf hingewiesen werden, dass die Nachricht irreversibel und für
+alle Gesprächsteilnehmer gelöscht wird. **\[\<=\]**
```

Comment:
Hier bräuchte man eventuell auch eine explizite Regelung für Redactions von Events, die geändert wurden. Wird nur das ursprüngliche Event redacted, verbleiben die Replacement-Events "lose" im Raum. Sie würden dann zwar nicht angezeigt werden wären Clients aber weiterhin zugänglich.

Um das zu beheben müsste der Client die ganze Kette durch mehrere Redactions entfernen. Alternativ gäbe es auch [MSC3912](https://github.com/matrix-org/matrix-spec-proposals/pull/3912).

Reply:
https://github.com/gematik/api-ti-messenger/pull/291/commits/3497a6e09eb9e6b72434799500a816228f533cb2